### PR TITLE
Axo-91 bug fix sga

### PR DESCRIPTION
**Pull request by shansfolder**
Fix the bug that sga will raise error when there is only one variant in the subgroup.

### CODE REVIEW

File: expan/core/experiment.py [hunk 1]
```diff
@@ -236,6 +236,21 @@ def sga(self, feature_name_to_bins, multi_test_correction=False):
 
         return subgroups
 
+    def _isNotValidForAnalysis(self, df):
```
Comment (gbordyugov) [inline expan/core/experiment.py]: Just a minor thing: why not `_isValidForAnalysis`?

One more general thing: I think that having a lot of methods that work on external dataframes goes a bit against the idea of encapsulation. The potential problem here is `df` is only particularly verified and all the checks that we run in the c'tor of `Experiment` are missing (please don't duplicate that code though :~)

Comment (shansfolder) [reply expan/core/experiment.py]: ```_isValidForAnalysis``` is a bit better. True.

Well, those general checks will still be run when we create an ```Experiment``` object regardlessly, so they are already verified. This method in this PR is an additional check when we are in ```sga```.

Comment (gbordyugov) [reply expan/core/experiment.py]: What I was saying is that no-one prevents you from calling `_isNotValidForAnalysis` with a totally random, `Experiment`-instance unrelated dataframe

Comment (gbordyugov) [reply expan/core/experiment.py]: But I guess it's a consequence of our design decision to split dataframes for SGA. Alternatively, we could spawn daughter instances of `Experiment` for each SGA sub-test.

Comment (shansfolder) [reply expan/core/experiment.py]: Got it. I recall the discussion now. We should keep track of technical debts.

**Comment by daryadedik** [comment]
looks good! thanks!

**Pull request merged**